build-script.py

```diff
@@ -189,7 +193,9 @@ def build(self, product_name, module_group_path=''):
       command.extend(['--product', product_name])
 
       env = dict(os.environ)
-      env['SWIFT_SYNTAX_CI_ENVIRONMENT'] = '1'
+      env['SWIFT_BUILD_SCRIPT_ENVIRONMENT'] = '1'
+      # Tell other projects in the unified build to use local dependencies
+      env['SWIFTCI_USE_LOCAL_DEPS'] = '1'
```

Comment:
Why do we need two environment variables instead of only one?

Reply:
My idea was that someone would still be able to build a non-build-script build (i.e. static libraries, no `enforce-exclusivity=unchecked`) while still using the already checked out sources. 

Similarly when building the stress tester you might want to create a CI-style build without manually checking out SwiftSyntax etc. You could do that by specifying `--no-local-dependencies` on the stress tester's build script.

It just feels like these are two different things that we should not mush together into a single flag.

Reply:
Ok, sounds good, but then should `SWIFTCI_USE_LOCAL_DEPS` have a more general name, like `SWIFT_BUILD_USE_LOCAL_DEPS` or something, and not seem like it should be only set by the CI system?

Reply:
The name needs to match the one used in the Swift package manager [here](https://github.com/apple/swift-package-manager/blob/879cb86615f5820990c5d802a55330c869731372/Package.swift#L268) to avoid it from pulling in remote dependencies. I could go and change it there first.

Reply:
The name needs to match the one used in the Swift package manager [here](https://github.com/apple/swift-package-manager/blob/879cb86615f5820990c5d802a55330c869731372/Package.swift#L268) to avoid it from pulling in remote dependencies. I could go and change it there first.

Reply:
It's not important, the exact spelling doesn't matter much

Reply:
OK. I’ll just leave it as is for now then.